car_cleaning_service_with_threads.rb

```diff
@@ -0,0 +1,115 @@
+require'time'
+
+class Car
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+
+  def to_s
+    @id
+  end
+
+  def to_sym
+    @id.to_sym
+  end
+end
+
+class CarCleaningService
+
+  MAX_CARS_AT_ONCE = 2
+  CAR_CLEANING_TIME = 2
+  ONE_HOUR_IN_SECONDS = 3600
+
+  attr_reader :time
+
+  def initialize(queue = Queue.new)
+    @queue = queue
+    @ready = {}
+    @processed_cars = 0
+    @time = Time.parse(Time.now.strftime("%Y-%m-%dT08:00:00%z"))
+    self.init_threads
+  end
+
+  public
+
+  def add_car(car)
+    @queue.push(car)
+    puts "[#{format_time}] Car #{car.to_s} queued for servicing. Estimated waiting time until car is serviced: #{waiting_time} hours."
+  end
+
+  def pick_up_car(car)
+    if @ready.has_key?(car.to_sym)
```

Comment:
you can add `ready` to `attr_reader` and use it without @ symbol. It will increase readability

Reply:
Thank you!